### DISCUSSION TITLE
巧用 Textbundle 格式让 Obsidian 更好的支持「包含图片子文件夹的 Markdown」

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2360] [服务]
## 重要提示

此方案适合 **[同步助手 1.0.1 版本](http://ksria.com/simpread/docs/#/Sync?id=%e4%b8%8b%e8%bd%bd)**，后续我会考虑在 1.0.2 版中原生增加此方案。

***

> 很多 Markdown 编辑器都支持 `xxx.md + assets/xxx.png` 的本地图片解决方案，如：Obsidian / Typora / Taio 等，如何让简悦也支持导出此种结构？

## Windows 用户

可以使用 Quicker 动作 [批量下载 Markdown 中的外链图片并替换成本地离线图片链接](https://github.com/Kenshin/simpread/discussions/2241)

## Mac 用户

> 利用 [同步助手导出 Textbundle](http://ksria.com/simpread/docs/#/Textbundle) 的方案，下面是步骤

1. 配置同步助手
2. 通过扩展端进入阅读模式 / 聚焦模式
3. 导出 `textbundle`
4. 导出后，将 `.textbundle` 后缀删除
5. 删除后，将会得到一个文件夹，进入此文件夹
6. 会看到 `xxx.markdown` 的结尾的文件，修改后缀为 `.md`

即可，然后此 `xxx.md` 与其对应的 `assets` 文件夹就可以在 **Obsidian / Typora / Taio** 使用了。

## 备注

Windows 用户也可使用此方式。

## 参考知识点

> 上述操作都已经写到了文档里面，有感兴趣的用户，可以看下。

- 简悦 · 同步助手的 [增强导出功能](http://ksria.com/simpread/docs/#/Sync?id=%e5%af%bc%e5%87%ba%e6%9c%8d%e5%8a%a1)
- [Textbundle](http://ksria.com/simpread/docs/#/Textbundle)

**Comment by byheaven** [comment]
非常需要这个功能，希望早日上线！
另外我的工作流一般是多个软件结合，比如全文保存在Obsidian中存档(base64编码会让md文件过大非常卡顿，因此最好能离线下载图片)，而标注会放到logseq里供写作时随时调用。所以希望能够支持分别设置保存的路径，谢谢！
还有图片文件夹的位置在每个软件里也不太一样，比如logseq就是一个单独的../assets文件夹，需要在里面直接放图片而不再建立新的文件夹层级；而Obsidian就没有类似的限制。

**Reply by Kenshin** [reply]
嗨，现在就可以使用的，只是需要先手动修改后缀。

**Reply by byheaven** [reply]
我的意思是能够自动化实现~比如：
1. 加入稍后读，自动保存全文为markdown文件，输出目录是Obsidian的Vault(同时可指定相应的图片储存目录)
2. 开始标注，自动保存标注到Logseq的目录。

**Reply by Kenshin** [reply]
标注的自动化支持导出到 Textbundle （当然这么做之后还得手动改名。

![image](https://user-images.githubusercontent.com/81074/130429931-a60f4b40-90a9-4765-badb-00958f2b6293.png)

同步助手下个版本会支持非 Textbundle 方案。

**Reply by fred1357944** [reply]
請問是每標註一次就會自動更新嗎？

**Reply by Kenshin** [reply]
> 請問是每標註一次就會自動更新嗎？

是的（需要使用自动化方案。）

**Comment by Kenshin** [comment]
嗨，各位。

简悦官方针对新用户或不喜欢配置的用户，发布了一个新功能：**Obsidan 配置库**。

细节请看下面，几分钟就能配置成完成。

***

![img](https://imgs.zhubai.love/c626678ff6b64dd1ac9dfdae6cf6d188.png)

> 针对新用户的一步到位的 Obsdian 配置库。

## 🗂 配置库

[简悦 · 配置库](https://www.yuque.com/kenshin/simpread/ds8zk0) 是简悦官方推出的一套针对新用户的极简配置方案，方便新用户用最快的方式使用简悦的各种高级服务，配置库内置了常用的双链笔记用法，如：Notion、Obsidian、Logseq、Roam Research，同时包含了简悦在阅读模式上的一些常规插件：Live Editor、题图、Safari 阅读模式等。

## 💡 特点

用户仅需要下载对应的配置库，修改里面的 UID 为自己的高级账户 UID，即可实现简悦的各种高级操作。

## 💁‍♂️ 受众

新用户或者不想使用手动配置的用户。

## 📝 如何使用

1️⃣ 下载

2️⃣ 修改为自己的 UID

3️⃣ 将必要的文件夹复制粘贴到相应位置

✅ Done！

## 📚 Obsidian 配置库

## 📝 特点

1️⃣ 加入稍后读，自动将本地快照和 Markdown 导入到 Obsidian

2️⃣ 加入标注时，自动将标注内容导入本地快照和 Obsidian

3️⃣ 改变稍后读或标注的元数据时，自动将改动的内容保存到本地快照和 Obsidian

4️⃣ 在 Obsidian 中直接内置好了简悦的标注系统，可直接在 Obsidian 实现标注

## 📖 区别

![img](https://imgs.zhubai.love/f4a51d8e8fb4401281b07f82bf0bc7d1.png)

## 📘 极速版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@little/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/xkuecp)

不需要同步助手，无任何配置，支持 Markdown 形式的快照，自动同步标注，适合 Obsidain + 简悦的尝鲜用户。

## 📙 基础版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@localrestapi/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/cg33gh)

需要同步助手，几乎无任何配置，支持本地快照，自动同步标注，适合 Obsidain + 简悦的轻量级用户。

## 📗 高级版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@sync/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/wq35mh)

需要同步助手，几乎无任何配置，支持本地快照，自动同步标注，可在 Obsidian 中标注，适合 Obsidain + 简悦的重度用户。
